script.js

```diff
@@ -0,0 +1,221 @@
+// User starts with either X or O
```

Comment:
Going to leave some inline comments throughout, but I think you already have an idea of the things I'll comment on about drying things up and refactoring.